activation/handler.go

```diff
@@ -232,7 +233,34 @@ func (h *Handler) SyntacticallyValidateDeps(
 		atx.NIPost,
 		expectedChallengeHash,
 		atx.NumUnits,
+		PostSubset([]byte(h.local)), // use the local peer ID as seed for random subset
```

Comment:
The reasons you brought up are exactly why I use peerID here :) 

First of all, there is no need for a publicly-verifiable randomness here. It's not required to verify if the given node checked the post labels it "was supposed" to check. The only requirement is that nodes verify slightly different random subsets (so that all nodes together verify the whole set). It's also OK for the seed to change.

Another thing, in a multi-smesher setup there is 1 gossip handler, which is not tied with any of the N identities the node works for. It doesn't validate incoming ATXs "on behalf" of any of these node IDs.

Reply:
I'm still unsure about this, this would also default to one node doing always validating the same amount of indices independent of how many identities it manages. While using the `NodeID` would automatically add more indices with more identities (because referencing a wrong ATX will cause multiple identities to be rendered invalid instead of just one).

Reply:
The node would verify the same number (k3) of indices no matter what the seed is. The operator is free to pick any K3 value that makes sense, including K3=K2.